src/bartiq/transform.py

```diff
@@ -0,0 +1,154 @@
+        if resource_name in expanded_aggregation_dict:
+            mapping = expanded_aggregation_dict[resource_name]
+            for sub_res, multiplier in mapping.items():
+                try:
+                    isinstance(multiplier, Number)
+                except TypeError:
+                    multiplier = backend.parse(multiplier)
+
+                if sub_res in aggregated_resources:
+                    current_value_expr = backend.parse(aggregated_resources[sub_res].value)
+                    aggregated_resources[sub_res].value = str(current_value_expr + multiplier * resource_expr)
+                else:
+                    new_resource = Resource(
+                        name=sub_res,
+                        type=subroutine.resources[resource_name].type,
+                        value=str(multiplier * resource_expr),
+                    )
+                    aggregated_resources[sub_res] = new_resource
+
+            del aggregated_resources[resource_name]
+
+    subroutine.resources = aggregated_resources
+    return subroutine
+
+
+def _expand_aggregation_dict(aggregation_dict: Dict[str, Dict[str, Any]], backend=BACKEND) -> Dict[str, Dict[str, Any]]:
+    """
+    Expand the aggregation dictionary to handle nested resources.
+    Args:
+        aggregation_dict: The input aggregation dictionary.
+    Returns:
+        Dict[str, Dict[str, Any]]: The expanded aggregation dictionary.
+    """
+    if not isinstance(aggregation_dict, dict):
+        raise TypeError("aggregation_dict must be a dictionary.")
+
+    expanded_dict = {}
+    for resource in aggregation_dict.keys():
+        expanded_dict[resource] = _expand_resource(resource, aggregation_dict, set())
+    return expanded_dict
+
+
+def _expand_resource(
+    resource: str, aggregation_dict: Dict[str, Dict[str, Any]], visited: set, backend=BACKEND
+) -> Dict[str, Any]:
+    """
+    Recursively expand resource mapping to handle nested resources and detect circular dependencies.
+    Args:
+        resource: The resource to expand.
+        aggregation_dict: The input aggregation dictionary.
+        visited: A set of currently visited resources to detect circular dependencies.
+    Returns:
+        Dict[str, Any]: The expanded resource mapping.
+    """
+    if resource in visited:
+        raise ValueError(f"Circular dependency detected: {' -> '.join(visited)} -> {resource}")
+
+    # Add current resource to the visited set
+    visited.add(resource)
+
+    # If the resource is not in the aggregation dictionary, return an empty dictionary
+    if resource not in aggregation_dict:
+        visited.remove(resource)
+        return {}
+
+    # Sympify the mapping values for the current resource
+    expanded_mapping = {
+        k: backend.parse(v) if not isinstance(v, Number) else k for k, v in aggregation_dict[resource].items()
+    }
+
+    expanded_mapping = {k: sympy.simplify(v) for k, v in aggregation_dict[resource].items()}
```

Comment:
We should not be using bare `sympy` anywhere here, always do everything symbolic through backend.
Also, as far as I understand, if we drop `simplify`, we'll just end up with slightly worse-looking expressions, so perhaps let's just remove it from here?